app/controllers/api/v1/registrant/auth_controller.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails5_api_controller_backport'
+require 'auth_token/auth_token_creator'
+
+module Api
+  module V1
+    module Registrant
+      class AuthController < ActionController::API
+        before_action :check_ip_whitelist
+
+        rescue_from(ActionController::ParameterMissing) do |parameter_missing_exception|
+          error = {}
+          error[parameter_missing_exception.param] = ['parameter is required']
+          response = { errors: [error] }
+          render json: response, status: :unprocessable_entity
+        end
+
+        def eid
```

Comment:
What does EID stand for? Is that in common use? Wouldn't it be better to use more meaningful name?

Reply:
It's eID, and yes, it's very common:

https://www.ria.ee/ee/eid-info-ja-juhendid.html
https://nortal.com/what-we-do/electronic-identity/

Reply:
`mobile_id` is a different case, the fact that we use `mid` is just wrong, as there is neither mID nor m-ID, the acronym is made up by us. We should use `mobile_id` or `mobiil_id`. 

I don't think `e_id` is an improvement over `eid`. Most of the usages do not have a dash or an underscore, the only implementation that I was able to find that uses e-ID instead of eID is [Malta](https://www.mita.gov.mt/en/eGov/eGovServices/Pages/e-ID.aspx). Estonia, Belgium, Finland, [Germany](https://www.bsi.bund.de/EN/Topics/ElectrIDDocuments/German-eID/german-eID_node.html) and [European Commission](https://ec.europa.eu/digital-single-market/en/e-identification) all use eID, and generally do not insert dashes or underscores, not even in URLs.

It's the case of `iPhone`, which as a Ruby method should be expressed either as `i_phone` or `iphone`. The first one can be confused with a beginning of a sentence "I phone my doctor get a sick leave", while the second one is most definitely a phone made by Apple corporation. Similar change in meaning happens with MasterCard. Another example of the same problem is `McDonalds`, which becomes either `mc_donalds` (probably a famous rapper, MC Donalds) or `mcdonalds`, a name of a fast-food restaurant. Ruby method names do not allow for conveying capitalisation nuances inside a word without changing it's meaning.

We could use `electronic_id`, but that's an artificial being, it is really uncommon, similarly to `electronic mail`, which is just `email`. Another option is `e_identification` or `e_identity_document`, `e_identity`. Unfortunately, in different countries `id` stands for different things, but the acronym is always common. 

`eid/eID` is the most commonly used version, we are just unfortunate that Ruby cannot represent it properly.